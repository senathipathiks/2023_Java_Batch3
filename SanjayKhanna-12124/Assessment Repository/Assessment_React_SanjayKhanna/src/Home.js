import React from 'react'

function Home() {
  return (
    <div>
      <h1>Welcome to the Laptop Allocation Management System!</h1>
      <div className='container'>
        <div className='c1'>
      <p>Welcome to the Laptop Allocation Management System (LAMS), a web-based application that allows you to request, assign, and track laptops and other IT assets for your employees.</p> 
      <p>LAMS is designed to simplify and streamline the process of IT asset management, from planning and ordering to operation and maintenance. With LAMS, you can:</p>
      <span className='sp'>- View the inventory of available laptops and other IT assets in real time</span>
      <span className='sp'>- Approve or reject requests from your subordinates or peers</span>
      <span className='sp'>- Assign a laptop or other IT asset to an employee or a host machine </span>
      <span className='sp'>- Track the location, status, and usage of each laptop or other IT asset</span>
      <span className='sp'>- Receive notifications and reminders for expiring licenses, end-of-life dates, and warranty details</span>
      <span className='sp'>- Retire or dispose of laptops or other IT assets that are no longer needed</span>
      <span className='sp'>- Generate reports and analytics for IT asset allocation and utilization</span><br/>
      <p>LAMS is integrated with the ServiceNow Asset Management product, which provides a consistent global process and a capability to track assets from end to end.</p>
      <p>LAMS also follows the best practices for assigning IT assets to employees, based on their level of seniority, role, and needs</p>
      <p>To get started, please log in with your credentials and select the appropriate option from the menu. If you have any questions or feedback, please contact the IT support team or visit the FAQ section. Thank you for using LAMS!</p>
    </div>
    </div>
    </div>
  )
}

export default Home