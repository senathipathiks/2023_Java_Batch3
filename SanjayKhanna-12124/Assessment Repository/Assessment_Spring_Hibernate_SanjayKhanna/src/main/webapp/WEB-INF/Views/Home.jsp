<%@page import="com.opms.bean.Petition"%>
<%@page import="java.util.ArrayList"%>
<%@ page language="java" contentType="text/html; charset=ISO-8859-1"
    pageEncoding="ISO-8859-1"%>
<!DOCTYPE html>
<html>
<head>
<meta charset="ISO-8859-1">
<title>Home Page</title>
<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.11.3/font/bootstrap-icons.min.css">
<style>
body{
background-image: url('https://media.licdn.com/dms/image/D4D12AQE4InJUdCby9g/article-cover_image-shrink_720_1280/0/1701464584710?e=2147483647&v=beta&t=kc8sbUtIDm6sl-g-_9HhJ7N1Ty9Md6aa3EYkJbAAWjg');
background-size: cover;
margin-left: 160px;
margin-top: 30px;
}
#display{
	border-radius: 5px;
  background-color: rgba(255,255,255);
  padding: 20px;
  width: 1200px;
  height: auto;
}
a{
padding-left: 10px;
}
</style>
</head>
<body>
<div id="display">
<% ArrayList<Petition> list = (ArrayList<Petition>)request.getAttribute("list");%>
<div class="text-center shadow p-3 bg-body rounded h3 mb-3">No of Record : <%=list.size()%></div>
<table class="table table-bordered table-hover text-center shadow">
 <thead>
    <tr class="table-primary">
      <th scope="col" >Petition Id</th>
      <th scope="col" >Category</th>
      <th scope="col" >Description</th>
      <th scope="col" >Priority</th>
      <th scope="col" >Raised By</th>
      <th scope="col" >Assigned To</th>
      <th scope="col" >Status</th>
      <th scope="col" class="col-3">Action</th>
    </tr>
  </thead>
 <tbody>
 <% 
  Petition v = (Petition)request.getAttribute("bean");
 
 for(Petition b : list){
	 if(v != null && v.getPetitionId()==b.getPetitionId()){
			out.print("<form action='SaveUpdate' method='post' modelAttribute='beanclass1'>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='petitionId' placeholder='petitionId' value='"+b.getPetitionId()+"'  readonly><label for='petitionId'>Petition Id</label></div></td>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='category' placeholder='Category' value='"+b.getCategory()+"' autocomplete='off'><label for='category'>Category</label></div></td>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='description' placeholder='Description'  value='"+b.getDescription()+"' autocomplete='off'><label for='description'>Description</label></div></td>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='priority' placeholder='Priority' value='"+b.getPriority()+"' autocomplete='off'><label for='priority'>Priority</label></div></td>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='raisedBy' placeholder='RaisedBy' value='"+b.getRaisedBy()+"' autocomplete='off'><label for='raisedBy'>Raised By</label></div></td>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='assignedTo' placeholder='Assigned To'  value='"+b.getAssignedTo()+"' autocomplete='off'><label for='assignedTo'>Assigned To</label></div></td>");
			out.print("<td><div class='form-floating'><input type='text' class='form-control' name='status' placeholder='Status' value='"+b.getStatus()+"' autocomplete='off'><label for='status'>Priority</label></div></td>");
			out.print("<td><button class='btn btn-success ms-2' type='submit' name='action' id='butt'><i class='bi bi-save'></i> save</button><a href='back'><button type='button' class='btn btn-success ' id='butt'><i class='bi bi-arrow-left'></i> Back</button></a></td></tr></form>");
	 }else{
	out.print("<tr><td>"+b.getPetitionId()+"</td><td>"+b.getCategory()+"</td><td>"+b.getDescription()+"</td><td>"+b.getPriority()+"</td><td>"+b.getRaisedBy()+"</td><td>"+b.getAssignedTo()+"</td><td>"+b.getStatus()+"</td><td><a href='findall?action=edit&vehicleId="+b.getPetitionId()+"' target='Operation'><button class='btn btn-primary'><i class='bi bi-pencil-square'></i> edit</button></a><a href='findall?action=delete&vehicleId="+b.getPetitionId()+"' target='Operation'><button class='btn btn-danger'><i class='bi bi-trash'></i>delete</button></a></td></tr>");
	 }
	 } 
	out.print("<form action='AddPetition' method='post' modelAttribute='beanclass2'>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='petitionId' placeholder='petitionId' disabled><label for='petitionId'>Petition Id</label></div></td>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='category' placeholder='Category' autocomplete='off'><label for='category'>Category</label></div></td>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='description' placeholder='Description' autocomplete='off'><label for='description'>Description</label></div></td>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='priority' placeholder='Priority' autocomplete='off'><label for='priority'>Priority</label></div></td>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='raisedBy' placeholder='RaisedBy' autocomplete='off'><label for='raisedBy'>Raised By</label></div></td>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='assignedTo' placeholder='Assigned To' autocomplete='off'><label for='assignedTo'>Assigned To</label></div></td>");
	out.print("<td><div class='form-floating'><input type='text' class='form-control' name='status' placeholder='Status'autocomplete='off'><label for='status'>Status</label></div></td>");
	out.print("<td><button class='btn btn-success ms-2' type='submit' name='action' id='butt'><i class='bi bi-plus-square'></i> add</button></td></tr></form>");
	 %>
 </tbody>
</table>
</div>
</body>
</html>