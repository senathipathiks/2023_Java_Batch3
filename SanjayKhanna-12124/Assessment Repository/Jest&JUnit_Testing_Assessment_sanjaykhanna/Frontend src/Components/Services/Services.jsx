
import axios from 'axios';
const origin = "http://localhost:1330";
const addPayee = "/addPayee"
const deletePayee = "/deletePayee?payeeId="
const updatePayee = "/updatePayee"
const getPayees = "/getPayees"
const getPayee = "/getPayee?payeeId="
const saveTransaction = "/saveTransaction"
const getTransactions = "/getTransactions"

const doAdd = (data) => {
    return axios.post(origin + addPayee, data);
}

const doDelete = (payeeId) => {
    return axios.delete(origin + deletePayee + payeeId);
}

const doUpdate = (data) => {
    return axios.put(origin + updatePayee, data);
}

const doGetPayee = (payeeId) => {
    return axios.get(origin + getPayee + payeeId);
}

const doGetAllPayee = () => {
    return axios.get(origin + getPayees);
}

const doSave = (data) => {
    return axios.post(origin + saveTransaction, data);
}

const doGetAllTransaction = () => {
    return axios.get(origin + getTransactions);
}

// exports.doAdd = doAdd;
// exports.doDelete = doDelete;
// exports.doGetAllPayee = doGetAllPayee;
// exports.doGetAllTransaction = doGetAllTransaction;
// exports.doGetPayee = doGetPayee;
// exports.doSave = doSave;
// exports.doUpdate = doUpdate;


export { doAdd, doDelete, doGetAllPayee, doGetAllTransaction, doGetPayee, doSave, doUpdate };

