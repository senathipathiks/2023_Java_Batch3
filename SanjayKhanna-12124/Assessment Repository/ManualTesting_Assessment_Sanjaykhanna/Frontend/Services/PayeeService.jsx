import axios from 'axios';
import { Component } from 'react';
const origin = "http://localhost:1330";
const addPayee = "/addPayee"
const deletePayee = "/deletePayee?payeeId="
const updatePayee = "/updatePayee"
const getPayees = "/getPayees"
const getPayee = "/getPayee?payeeId="

class PayeeService extends Component {
 
    doAdd = (data) =>{
        return axios.post(origin+addPayee,data);
    }

    doDelete = (payeeId) =>{
        return axios.delete(origin+deletePayee+payeeId);
    }

    doUpdate = (data) => {
        return axios.put(origin+updatePayee,data);
    }

    doGetPayee = (payeeId) =>{
        return axios.get(origin+getPayee+payeeId);
    }

    doGetAll = () => {
        return axios.get(origin+getPayees);
    }
}

const payeeService = new PayeeService();
export default payeeService;
