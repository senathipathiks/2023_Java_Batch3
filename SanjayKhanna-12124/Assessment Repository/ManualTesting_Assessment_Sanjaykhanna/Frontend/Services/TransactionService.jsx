import axios from 'axios';
import { Component } from 'react';
const origin = "http://localhost:1330";
const saveTransaction = "/saveTransaction"
const getTransactions = "/getTransactions"
class TransactionService extends Component {
  
    doSave = (data) => {
        return axios.post(origin+saveTransaction,data);
    }

    doGetAll = () =>{
        return axios.get(origin+getTransactions);
    }
}
const transactionService = new TransactionService();
export default transactionService;
