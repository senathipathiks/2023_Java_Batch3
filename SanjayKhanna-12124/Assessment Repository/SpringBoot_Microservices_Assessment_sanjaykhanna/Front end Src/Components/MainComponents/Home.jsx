import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import Loading from '../UtilitiesComponents/Loading';

const Home = () => {

    const navigate = useNavigate();

    const[flag,setFlag] = useState();

    const handleClick = (e) =>{
        e.preventDefault();
        setFlag(true);
        setTimeout(()=>{navigate('/product')},2000)
    };

    return (
        <div id='home' className='container mt-5'>
            <div className='card w-75 shadow mx-auto'>
                <div className='card-title text-center  mt-5 mb-3'>
            <h1>Welcome to the home page!</h1></div>
            <div className='card-body ms-5 h5'>
            <p>This is a simple website created using HTML and CSS. The purpose of this site is to provide an example for beginners</p>
            <p>This is a simple application that showcases some of my skills in web development.  The main features include:</p>
            <p>This is a simple example of a Landing Page site built with React.js.</p> 
            <p>This is a Ssimple website for Inventory Management</p>
            <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis nihil sequi praesentium ratione nostrum exercitationem, sapiente odit temporibus, dicta enim mollitia cum vitae maxime unde laborum repudiandae quibusdam libero assumenda!</p>
            </div>
            <div className='card-footer border-0 text-center'>
            <button className="btn btn-primary" onClick={(e)=>handleClick(e)}>Get Started</button>
            </div>
            </div>
            <Loading  flag={flag}/>
        </div>
    );
}

export default Home;
