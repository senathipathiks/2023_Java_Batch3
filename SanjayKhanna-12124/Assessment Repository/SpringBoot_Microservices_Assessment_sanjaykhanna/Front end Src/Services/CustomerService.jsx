import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createCustomer = "createcustomer";
const readAllCustomer = "readallcustomer";
const updateCustomer = "updatecustomer";
const deleteCustomer = "deletecustomer?id=";
const readCustomer = "readcustomer?id=";

class CustomerService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createCustomer, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllCustomer);
    };

    doRead = (id) => {
        return axios.get(origin + readCustomer + id);
    };

    doUpdate = (data) => {
        return axios.put(origin + updateCustomer, data);
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteCustomer + id)
    };
}
const customerService = new CustomerService();
export default customerService;
