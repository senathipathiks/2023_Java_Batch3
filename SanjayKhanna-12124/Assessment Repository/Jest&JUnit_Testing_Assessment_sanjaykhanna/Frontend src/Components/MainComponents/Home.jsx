import { Button } from '@mui/material';
import React, { useEffect, useState } from 'react';
import { doGetAllPayee } from '../Services/Services';
import Search from '../UtilitiesComponents/Search';
import AddPayee from './AddPayee';
import DeletePayee from './DeletePayee';
import EditPayee from './EditPayee';

const Home = () => {

    const [data, setData] = useState([]);
    const [eId, setEventId] = useState(0);
    const [search, setSearch] = useState("");

    const [Flags, setFlags] = useState({
        isAddable: false,
        isEditable: false,
        isDeletable: false
    });

    useEffect(()=>{
        doGetAllPayee().then(res=>{
            setData(res.data);
        })
        .catch(err=>{
            console.log(err);
        })
    },[])

    return (
        <div>
            <section className="container container-lg">
                <div className="card card-lg mt-5 shadow-lg p-2 mx-auto">
                    <div className="card-header card-title border-0 text-center h1" title='homeHeader' data-testid='homeHeader'>List of Payee</div>
                    <div className="card-body">
                        <div className="container row mt-3">
                            <div className="col-6" style={{ "textAlign": "left" }}>
                                <Button variant="contained" data-testid='addPayee' onClick={() => { setFlags({ isAddable: true }) }} color="success">Add Payee</Button></div>
                            <div className="col-6">
                                <Search search={search} setSearch={setSearch} />
                            </div>
                        </div>
                        <div className='container' title='payeeTable'>
                            <table className="table table-bordered table-striped table-hover text-center mt-3">
                                <thead className="bg-secondary text-white">
                                    <tr>
                                        <td>Payee Name</td>
                                        <td>Nick Name</td>
                                        <td>Account Number</td>
                                        <td className="col-2">Action</td>
                                    </tr>
                                </thead>
                                <tbody>
                                    {
                                        data.filter((st) => st.payeeName.toLowerCase().includes(search) || st.payeeName.toUpperCase().includes(Search)).map((item) => (
                                            <tr key={item.payeeId}>
                                                <td>{item.payeeName}</td>
                                                <td>{item.nickName}</td>
                                                <td>{item.account.accountNumber}</td>
                                                <td className="d-flex justify-content-evenly"><Button variant="contained" onClick={() => { setFlags({isEditable:true});setEventId(item.payeeId)}}>edit</Button><Button variant="contained" color="error" onClick={() => { setFlags({isDeletable:true});setEventId(item.payeeId) }}>delete</Button></td>
                                            </tr>
                                        ))
                                    }
                                </tbody>
                            </table>
                        </div>
                    </div>
                </div>
            </section>
            {Flags.isAddable && <AddPayee Flags={Flags} />}
            {Flags.isEditable && <EditPayee Flags={Flags} id={eId} />}
            {Flags.isDeletable && <DeletePayee Flags={Flags} id={eId} />}
        </div>
    );
}

export default Home;
