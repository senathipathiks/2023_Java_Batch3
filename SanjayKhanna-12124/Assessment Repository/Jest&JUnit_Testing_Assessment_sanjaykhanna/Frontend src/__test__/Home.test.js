import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../Components/MainComponents/Home';
import { doGetAllPayee } from '../Components/Services/Services';

describe('Checking for Home Page Rendering', () => {

  test('should render home page', () => {
    render(<Home />)
    expect(screen.getByText('List of Payee')).toBeInTheDocument();
  });

  test('should check the header', () => {
    render(<Home />)
    expect(screen.getByTestId('homeHeader')).toBeInTheDocument();
    expect(screen.getByTestId('homeHeader')).toHaveTextContent('List of Payee');
  });

  test('should check rendering the table', () => {
    render(<Home />)
    expect(screen.getByTitle('payeeTable')).toBeInTheDocument();
  });

  test('should return payee list', () => {
    const result = doGetAllPayee();
    expect(result).toBeDefined();
    result.then(res => {
      expect(res.status).toEqual(200);
      expect(res.data).toEqual([]);
    });
  });

  test('Snapshot Testing for Home Page', () => {
    expect(renderer.create(<Home />).toJSON()).toMatchSnapshot();
  });
  

});



