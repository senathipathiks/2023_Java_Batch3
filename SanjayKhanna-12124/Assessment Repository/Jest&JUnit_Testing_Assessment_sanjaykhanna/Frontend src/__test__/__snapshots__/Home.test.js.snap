// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking for Home Page Rendering Snapshot Testing for Home Page 1`] = `
<div>
  <section
    className="container container-lg"
  >
    <div
      className="card card-lg mt-5 shadow-lg p-2 mx-auto"
    >
      <div
        className="card-header card-title border-0 text-center h1"
        data-testid="homeHeader"
        title="homeHeader"
      >
        List of Payee
      </div>
      <div
        className="card-body"
      >
        <div
          className="container row mt-3"
        >
          <div
            className="col-6"
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-ke5b6m-MuiButtonBase-root-MuiButton-root"
              data-testid="addPayee"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Add Payee
            </button>
          </div>
          <div
            className="col-6"
          >
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r0:"
                  name="search"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search..."
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-ihdtdm"
                  >
                    <span
                      className="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          className="container"
          title="payeeTable"
        >
          <table
            className="table table-bordered table-striped table-hover text-center mt-3"
          >
            <thead
              className="bg-secondary text-white"
            >
              <tr>
                <td>
                  Payee Name
                </td>
                <td>
                  Nick Name
                </td>
                <td>
                  Account Number
                </td>
                <td
                  className="col-2"
                >
                  Action
                </td>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </section>
</div>
`;
